### PR TITLE
Fix/check 267 show modal for existing source

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: src/app/helpers.js [hunk 1]
```diff
@@ -240,6 +240,15 @@ function getErrorMessageForRelayModernProblem(errorOrErrors) {
   return null;
 }
 
+function getErrorObjectForRelayModernProblem(errorOrErrors) {
```
Comment (amoedoamorim) [inline src/app/helpers.js:243]: Please make it `getErrorObjectsForRelayModernProblem` for consistency.

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Looks great, Sawy! Added just a couple notes.

File: src/app/components/media/SetSourceDialog.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    </Dialog>
+  );
+}
+
+SetSourceDialog.propTypes = {
+  open: PropTypes.bool.isRequired,
+  sourceName: PropTypes.string.isRequired,
+  primaryUrl: PropTypes.string.isRequired,
+  onCancel: PropTypes.func.isRequired, // func() => undefined
+  onSubmit: PropTypes.func.isRequired, // func(<Project>) => undefined
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/media/SetSourceDialog.js:90]: no will updated it

File: src/app/helpers.js (hunk 1 above)
Comment (melsawy) [reply src/app/helpers.js:243]: Done

File: src/app/components/media/SetSourceDialog.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            description="Dialog title for existing source with same name"
+          />
+        }
+      </DialogTitle>
+      <DialogContent>
+        <Typography>
+          { primaryUrl ?
+            <FormattedHTMLMessage
+              id="setSourceDialog.existDescriptionWithUrl"
+              defaultMessage="An the source <b>{name}</b> with the primary URL <b>{url}</b> already exists."
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/media/SetSourceDialog.js:40]: This copy send by Pierre https://www.figma.com/proto/S7gC137KOHCYwDpZIe07DD/Sources?node-id=54%3A2829&viewport=1750%2C453%2C0.15321724116802216&scaling=min-zoom

File: src/app/components/media/SetSourceDialog.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+              values={{
+                name: sourceName,
+                url: primaryUrl,
+              }}
+              description="Text to inform user about existing source"
+            /> :
+            <FormattedHTMLMessage
+              id="setSourceDialog.existDescription"
+              defaultMessage="An the source <b>{name}</b> already exists."
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/media/SetSourceDialog.js:49]: This copy send by Pierre https://www.figma.com/proto/S7gC137KOHCYwDpZIe07DD/Sources?node-id=54%3A2829&viewport=1750%2C453%2C0.15321724116802216&scaling=min-zoom

**Pull request merged**